lib/content-services/src/lib/search/components/search-properties/search-properties.component.ts

```diff
@@ -0,0 +1,216 @@
+    }
+
+    getExtensionWithoutDot(extension: string): string {
+        const extensionSplitByDot = extension.split('.');
+        return extensionSplitByDot[extensionSplitByDot.length - 1];
+    }
+
+    filterExtensions = (extensions: string[], filterValue: string): string[] => {
+        const filterValueLowerCase = this.getExtensionWithoutDot(filterValue).toLowerCase();
+        const extensionWithDot = filterValue.startsWith('.');
+        return extensions.filter((option) => {
+            const optionLowerCase = option.toLowerCase();
+            return extensionWithDot && filterValueLowerCase ? optionLowerCase.startsWith(filterValueLowerCase) : optionLowerCase.includes(filterValue);
+        });
+    };
+
+    reset() {
+        this.form.reset();
+        this.reset$.next();
+        this.displayValue$.next('');
+    }
+
+    submitValues() {
+        if (this.settings && this.context) {
+            let query = '';
+            let displayedValue = '';
+            if (this.form.value.fileSize !== undefined && this.form.value.fileSize !== null) {
+                displayedValue = `${this.translateService.instant(this.form.value.fileSizeOperator)} ${this.form.value.fileSize} ${this.translateService.instant(this.form.value.fileSizeUnit.abbreviation)}`;
+                const size = this.form.value.fileSize * this.form.value.fileSizeUnit.bytes;
+                switch (this.form.value.fileSizeOperator) {
+                    case FileSizeOperator.AT_MOST:
+                        query = `${this.sizeField}:[0 TO ${size}]`;
+                        break;
+                    case FileSizeOperator.AT_LEAST:
+                        query = `${this.sizeField}:[${size} TO MAX]`;
+                        break;
+                    default:
+                        query = `${this.sizeField}:[${size} TO ${size}]`;
+                }
+            }
+            if (this._selectedExtensions?.length) {
+                if (query) {
+                    query += ' AND ';
+                    displayedValue += ', ';
+                }
+                query += `${this.nameField}:("*.${this._selectedExtensions.join('" OR "*.')}")`;
+                displayedValue += this._selectedExtensions.join(', ');
+            }
+            this.displayValue$.next(displayedValue);
+            this.context.queryFragments[this.id] = query;
+            this.context.update();
+        }
+    }
+
+    hasValidValue(): boolean {
+        return true;
+    }
+
+    getCurrentValue(): SearchProperties {
+        return {
+            fileSizeCondition: this.form.getRawValue(),
+            fileExtensions: this._selectedExtensions
+        };
+    }
+
+    setValue(searchProperties: SearchProperties) {
+        this.form.patchValue(searchProperties.fileSizeCondition);
+        this.selectedExtensions = searchProperties.fileExtensions;
+        this.submitValues();
+    }
+
+    private getOperatorNameWidth(operator: string, font: string): number {
```

Comment:
Might be obsolete if we decide on setting `max-width`, same for other methods lower

Reply:
Do you have any other suggestion for naming? 
That function takes some text (operator name) like 'At Least' and calculates actual width which that text takes. 
Text has not 'max-width', text has just some width which it takes.

Reply:
No no no, the comment was about potential removal of those method if we wanted to set `max-width` instead of calculating the width but we're doing this now so you can ignore this comment